nuts/helpers/filters.py

```diff
@@ -2,12 +2,51 @@
 Functions to filter the nornir inventory and used in conjunction with
 a context's nornir_filter function.
 """
-from typing import Optional, Dict, Any, List, Set
+from typing import Optional, Dict, Any, List, Union
+from nornir.core.filter import F, OR
 
-from nornir.core.filter import F
+from nuts.helpers.errors import NutsSetupError
 
 
 def filter_hosts(test_data: Optional[List[Dict[str, Any]]]) -> F:
     assert test_data is not None
-    hosts: Set[str] = {entry["host"] for entry in test_data}
+    hosts: List[str] = list({entry["host"] for entry in test_data})
```

Comment:
Is it desired to have the same host multiple times in the same list. How is it prevented now?

Reply:
It is still a set comprehension but in the end, it is converted to a list for consistency.